Source/WebCore/html/NavigatorUserActivation.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2022 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "Supplementable.h"
+#include <wtf/Forward.h>
+
+namespace WebCore {
+
+class UserActivation;
+class Navigator;
+
+class NavigatorUserActivation final : public Supplement<Navigator> {
+    WTF_MAKE_FAST_ALLOCATED;
+public:
+    explicit NavigatorUserActivation(Navigator&);
```

Comment:
Does the constructor need to be public?

Reply:
Yeah, otherwise it (guessing binding layer) doesn't like it:

```
/Volumes/Data/dev/OpenSource/WebKitBuild/Release/usr/local/include/wtf/StdLibExtras.h:544:17: note: in instantiation of function template specialization 'std::make_unique<WebCore::NavigatorUserActivation, WebCore::Navigator &>' requested here
    return std::make_unique<T>(std::forward<Args>(args)...);
In file included from /Volumes/Data/dev/OpenSource/WebKitBuild/Release/DerivedSources/WebCore/unified-sources/UnifiedSource186.cpp:6:
In file included from ./html/NavigatorUserActivation.cpp:27:
./html/NavigatorUserActivation.h:45:14: note: declared private here
    explicit NavigatorUserActivation(Navigator&);
```